tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -125,6 +125,7 @@ internal TracerManager CreateTracerManager(ImmutableTracerSettings settings, Tra
             }
 
             var gitMetadataTagsProvider = GetGitMetadataTagsProvider(settings, scopeManager);
+            var extraServiceProvider = new ExtraServicesProvider();
```

Comment:
Using this approach will replace the provider every time the customer reconfigures the tracer. That _may_ not be what you want? 🤔 Would you prefer a "singleton"-esque approach, so the service list persists across reconfigures? Or should we reset the service list each time as you are now. Given a reconfigure _could_ change the service list, maybe that's ok 🤔

Reply:
Made the service a singleton, also removes the need to pass it round so much, see 14f828b6ca89a2f776967e35a647b6493ec40c97